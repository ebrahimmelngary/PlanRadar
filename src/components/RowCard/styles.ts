import { StyleSheet } from "react-native";
import { Colors } from "../../constant/colors";

export default StyleSheet.create({
    rowCard: {
        width: '100%',
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingVertical: 8,
        alignItems: 'center',
        paddingHorizontal: 16,
    },
    rowLabel: {
        fontSize: 14,
        color: Colors.black,
        fontWeight: 'bold',
    },
    rowValue: {
        fontSize: 20,
        fontWeight: 'bold',
        color: Colors.primary,
    },
});