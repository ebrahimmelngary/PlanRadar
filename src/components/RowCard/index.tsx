import { Text, View } from 'react-native';
import styles from './styles';

export const RowCard = ({ label, value }: { label: string; value: string }) => {
  return (
    <View style={styles.rowCard}>
      <Text style={styles.rowLabel}>{label}</Text>
      <Text style={styles.rowValue}>{value}</Text>
    </View>
  );
};
