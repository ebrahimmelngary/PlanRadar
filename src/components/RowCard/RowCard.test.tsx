import React from 'react';
import { render } from '@testing-library/react-native';
import { RowCard } from '../RowCard';

describe('RowCard Component', () => {
  it('renders label and value correctly', () => {
    const { getByText } = render(<RowCard label="Temperature" value="25°C" />);

    expect(getByText('Temperature')).toBeTruthy();
    expect(getByText('25°C')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(<RowCard label="Humidity" value="60%" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
