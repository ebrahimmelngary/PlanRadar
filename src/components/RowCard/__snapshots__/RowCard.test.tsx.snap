// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowCard Component matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Humidity
  </Text>
  <Text
    style={
      {
        "color": "#2388C7",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    60%
  </Text>
</View>
`;
